Feature/student course enrollment

nataliazm99 opened the pull request:
This PR creates a view to gather the necessary data to replicate the[ Student Enrollment dashboard](https://us-east-1.online.tableau.com/t/codeorg/views/EnrollmentDemographics/EnrollmentDemographics) in Tableau. 

There is a difference worth noting to the current enrollment dashboard: the new view doesn't include HOC activity because 1) it is pulling from `dim_user_course_activity` as this is a dashboard aggregating by course and school year. 2) I think it is good not to include HOC because we only have HOC activity for signed-in users so it can be confusing/misleading to display it.

<!--
changelog:
auth.      descr.      date
natalia        init        02-07-2025 
-->

coryamanda commented:
Questions @nataliazm99 :
- All the coalesce statements on school fields. This substitutes 0 for NULL values for the school features. That'll result in conflicts with Trevor if they look at the same data by title 1 status, frl eligible percent, etc. which will still have null values. What's the reaon to group null with 0 in this case if not overall?
- I'm assuming the grouping step at the end is not actually part of it, right? Should I just drop that?

coryamanda commented:
Second question, which I wrote and then deleted, and now I'm writing again. All the coalesce steps on the schools CTE - why take away all the null values from these? Won't this lead to conflicting results with Trevor?

coryamanda commented:
Third question - section size isn't in the final table, is there a reason you create it and then drop it?

nataliazm99 commented:
@coryamanda for your comments above:
- I use coalesces on school characteristics because this view is for replacing the student enrollment dashboard, and the school characteristics are used in [this tab](https://us-east-1.online.tableau.com/t/codeorg/views/EnrollmentDemographics/Enrollment-AFECriteria) to get to activity in underserved schools. Schools for which their school characteristics are null are mostly non-NCES schools, and if we don't have data for some of these characteristics they should not be counted as underserved. For these aggregations, adding 'unknown/null' in the filters adds unnecessary complexity for the user not needed to identify underserved schools.  This view won't be exposed in Trevor, this viz doesn't display individual schools; the last tab does but doesn't display school characteristics, so I don't think there's an issue with treating it differently than in other models. I can add a note to the dashboard explaining that nulls for school characteristics are treated as a No.
- Thanks for catching the section size, I added it.
- Same for the testing query at the end, I removed it.

coryamanda reviewed (approved):
comments addressed, looks good to me
